src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/LogicalExpression.java

```diff
@@ -41,7 +75,10 @@ public void traverseForNot(LogicalExpression parent) {
     while (iterator.hasNext()) {
       LogicalExpression logicalExpression = iterator.next();
       if (logicalExpression.logicalRelation == LogicalOperator.NOT) {
+        // TODO: why are we removing the element from the list ?
         iterator.remove();
+
+        // TODO wy use this and why not use -=
```

Comment:
Can definitely change to use `-=` operator to simplify statement, probably just an oversight.

Reply:
Yuqi Resolve: changed